### ISSUE TITLE
Limit not returning what I expected

### ISSUE THREAD

**Issue by asantaga** [tchellomello/python-ring-doorbell#45]
Hi there,

Slightly confused at what the limit operation does.. I expected it to limit the number of records returned but it seams to be indicating how many records to query from the api , regardless of type.

eg.
```python
from ring_doorbell import Ring
myring = Ring('xxxxx@gmail.com', 'yyyy')
print ("did we connect %s " % myring.is_connected)
for bells in list(myring.doorbells):
    for event in bells.history(limit=15,kind='ding'):
        print('ID:       %s' % event['id'])
        print('Kind:     %s' % event['kind'])
        print('Answered: %s' % event['answered'])
        print('When:     %s' % event['created_at'])
        print('--' * 50)
```

I expected to return the last 15 "dings" but it returns nothing
if I change the limit to 1000 then I get a couple but not 5-10 or so..

**Comment by tchellomello** [comment]
Hello @asantaga , 
That is a bug that we have to enhance. Basically what happens is that the first the limit is applied and then the kind. So for example, if you have a total of 100 events and just 5 are `ding` and let's say the dings are the latest 5 events, it will just show if you do limit=100, kind='ding'.
I think we can modify the API to select all events and keep searching until the limit + kind criteria get met. 
I'm acknowledging this bug and let's work to fix it. 
Thanks for reporting it!
mmello

**Comment by asantaga** [comment]
@tchellomello pleasure, yeah the fix would be to keep executing the query until the number of "kind" are found, OR a date limit reached..  You obviously dont want to keep querying events for a long time .. 
thanks

**Comment by tchellomello** [comment]
@asantaga  I've published the PR #56  that fixes this issue.

I would appreciate if you could test or provide any feedbacks.

**Comment by tchellomello** [comment]
Implemented by PR #56
